acasclient/ddict.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import unicode_literals
+from typing import Any, Dict
+
+import logging
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class DDict(object):
+    """The DDict class is meant as a generic interface for any implementation of a Data Dictionary that
+    can be referenced by an ACAS CodeValue.
+    Any classes implementing this interface must implement the get_values() method.
+    """
+
+    def __init__(self, code_type, code_kind, code_origin):
+        self.code_type = code_type
+        self.code_kind = code_kind
+        self.code_origin = code_origin
+        self.valid_values = None
+
+    def get_values(self):
```

Comment:
Agree on having `client` for consistency. I reworked a little how this gets populated so it is getting called from outside this class. In that case I'll rename it to `update_valid_values` since it is a public method.